Release_Process.md

```diff
@@ -59,3 +59,12 @@ These steps are common between minor and patch releases:
         (without the 'v' in front) is _not_ active.
      3. Make sure the **stable** branch is _not_ active.
      4. Scroll to the bottom of the page and click the Submit button.
+1. Update all deployments with the currently released version accordingly.
```

Comment:
It seems like this point is too late in the release process to be updating the BigchainDB version number in `k8s/bigchaindb/bigchaindb-dep.yaml`. Shouldn't that be done around the same time as the update of the version number in `bigchaindb/version.py` (before the release on GitHub and PyPI)?